File: app/src/main/java/com/shmedo/mcloudapp/base/BaseActivity.java
package com.shmedo.mcloudapp.base;

import android.os.Build;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import com.shmedo.mcloudapp.util.XPermissionUtils;
import com.shmedo.mcloudapp.util.common.HandleBackUtil;

import butterknife.ButterKnife;
import timber.log.Timber;

/**
 * 项目名：  mCloudapp
 * 包名：    com.shmedo.mcloudapp.base
 * 文件名:   BaseActivity
 * 创建者:   dpc
 * 创建时间:  2019/1/8 09:46
 * 描述：    TODO
 */
public abstract class BaseActivity extends AppCompatActivity {

    protected abstract int initContentView();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //StatusUtil.StatusBarLightMode(this);
        setContentView(initContentView());
        initState();
        ButterKnife.bind(this);
    }


    /**
     * 沉浸式状态栏
     */
    public void initState() {
        //设置了Theme带ActionBar的Activity，左上角显示返回按钮
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            if (Build.VERSION.SDK_INT >= 21) {
                getSupportActionBar().setElevation(0);
            }
        }
        /*if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            //透明状态栏
            getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
            //透明导航栏
            getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
        }*/
    }


    @Override
    protected void onResume() {
        super.onResume();

        String name = getClass().getName();
        Timber.d("startPage,activity=" + name);
    }


    @Override
    protected void onPause() {
        super.onPause();

        String name = getClass().getName();
        Timber.d("endPage,activity=" + name);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }



    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull
            int[] grantResults) {
        XPermissionUtils.onRequestPermissionsResult(requestCode, permissions, grantResults);
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }


    @Override public void onBackPressed() {
        if (!HandleBackUtil.handleBackPress(this)) {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/com/shmedo/mcloudapp/ui/page/base/activity/BaseActivity.kt
package com.shmedo.mcloudapp.ui.page.base.activity

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import com.blankj.utilcode.util.KeyboardUtils

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2022/11/28 <br></br>
 * 描述：     TODO
 */
abstract class BaseActivity : BaseVmDbActivity() {

    override fun initViewModel() {}

    override fun initView(savedInstanceState: Bundle?) {}

    override fun initData() {}

    /**
     * 创建liveData观察者
     */
    override fun createObserver() {}

    /**
     * Use a Toolbar as an Action Bar
     */
    open fun setToolBar(toolbar: Toolbar) {
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = ""
    }

    open fun hideTitleBar() {
        supportActionBar?.hide()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        //ActionBar Home按钮返回事件
        if (item.itemId == android.R.id.home) {
            KeyboardUtils.hideSoftInput(this.window.decorView)
            onBackPressedDispatcher.onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}